internal/app/auth_server.go

```diff
@@ -0,0 +1,226 @@
+			CommonName:         cfg.Config.Auth.DN.CN,
+			Organization:       cfg.Config.Auth.DN.O,
+			OrganizationalUnit: cfg.Config.Auth.DN.OU,
+			Country:            cfg.Config.Auth.DN.C,
+			Locality:           cfg.Config.Auth.DN.L,
+			Province:           cfg.Config.Auth.DN.ST,
+			StreetAddress:      cfg.Config.Auth.DN.Street,
+			PostalCode:         cfg.Config.Auth.DN.PostalCode,
+		}
+	}
+
+	storage := appauth.NewAuthStorage(
+		&appauth.StorageConfig{
+			Domain: domain.String(),
+			Debug:  cfg.Debug,
+			DN:     dn,
+		},
+		userUsecase.GetUserBySubject,
+	)
+	handler, err := op.NewOpenIDProvider(
+		ctx,
+		config,
+		storage,
+		op.WithHttpInterceptors(jsonToFormHandler()),
+		op.WithHttpInterceptors(setURLVarsHandler()),
+		op.WithCustomKeysEndpoint(op.NewEndpoint(".well-known/jwks.json")),
+	)
+	if err != nil {
+		e.Logger.Fatal(err)
+	}
+
+	router := handler.HttpHandler().(*mux.Router)
+
+	if err := router.Walk(muxToEchoMapper(r)); err != nil {
+		e.Logger.Fatal(err)
+	}
+
+	// Actual login endpoint
+	r.POST("api/login", login(ctx, cfg, storage, userUsecase))
+
+}
+
+func setURLVarsHandler() func(handler http.Handler) http.Handler {
+	return func(handler http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path != "/authorize/callback" {
+				handler.ServeHTTP(w, r)
+				return
+			}
+
+			r2 := mux.SetURLVars(r, map[string]string{"id": r.URL.Query().Get("id")})
+			handler.ServeHTTP(w, r2)
+		})
+	}
+}
+
+func jsonToFormHandler() func(handler http.Handler) http.Handler {
+	return func(handler http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path != "/oauth/token" {
+				handler.ServeHTTP(w, r)
+				return
+			}
+
+			if r.Header.Get("Content-Type") != "" {
+				value, _ := header.ParseValueAndParams(r.Header, "Content-Type")
+				if value != "application/json" {
+					// Content-Type header is not application/json
+					handler.ServeHTTP(w, r)
+					return
+				}
+			}
+
+			if err := r.ParseForm(); err != nil {
+				return
+			}
+
+			var result map[string]string
+
+			if err := json.NewDecoder(r.Body).Decode(&result); err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				return
+			}
+
+			for key, value := range result {
+				r.Form.Set(key, value)
+			}
+
+			handler.ServeHTTP(w, r)
+		})
+	}
+}
+
+func muxToEchoMapper(r *echo.Group) func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error {
+	return func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error {
+		path, err := route.GetPathTemplate()
+		if err != nil {
+			return err
+		}
+
+		methods, err := route.GetMethods()
+		if err != nil {
+			r.Any(path, echo.WrapHandler(route.GetHandler()))
+			fmt.Println("ANY| " + path)
+			return nil
+		}
+
+		for _, method := range methods {
+			r.Add(method, path, echo.WrapHandler(route.GetHandler()))
+			fmt.Println(method + "| " + path)
```

Comment:
Replace ALL fmt.Println to log.Infof and add prefix to log message: "auth"
```suggestion
			log.Infof("auth: %s %s", method, path)
```

Note: log is from reearth-backend/pkg/log

Reply:
Those ware for debugging, I removed them